crates/bevy_ui/src/node_bundles.rs

```diff
@@ -158,6 +188,31 @@ pub struct AtlasImageBundle {
     pub z_index: ZIndex,
 }
 
+impl Clone for AtlasImageBundle {
+    fn clone(&self) -> Self {
+        // A `MeasureFunc` is not cloneable.
+        assert!(
+            self.calculated_size.measure_func.is_none(),
+            "AtlasImageBundle with non-default `ContentSize` is not cloneable"
+        );
```

Comment:
I think `Clone` should never panic. As a user I would not expect that could happen, though the Rust documentation does not say that, probably because cloning is done explicitly.
I'd suggest to not implement that trait and instead provide a method `try_clone(&self) -> Option<Self>` or something like `Result<Self, MeasureFuncIsSome>`.

Reply:
I generally like this idea, it will let the application provide a mechanism to deal with error instead of just crashing. If the pattern becomes common we could create a `TryClone` trait

Reply:
Yep, a try_clone method returning a result makes more sense.